dashboard/app/controllers/lti_v1_controller.rb

```
@@ -87,13 +89,26 @@ def authenticate
       user = Queries::Lti.get_user(decoded_jwt)
       target_link_uri = decoded_jwt[:'https://purl.imsglobal.org/spec/lti/claim/target_link_uri']
 
+      launch_context = decoded_jwt[Policies::Lti::LTI_CONTEXT_CLAIM]
+      nrps_url = decoded_jwt[Policies::Lti::LTI_NRPS_CLAIM][:context_memberships_url]
+      resource_link_id = decoded_jwt[Policies::Lti::LTI_RESOURCE_LINK_CLAIM][:id]
+      deployment_id = decoded_jwt[Policies::Lti::LTI_DEPLOYMENT_ID_CLAIM]
+      deployment = Queries::Lti.get_deployment(integration.id, deployment_id)
```

What if the deployment_id isn't found? Like maybe we forgot a setup step and didn't add it. Will the teacher receive an error message they can share with us so we can quickly figure this out?

Or maybe we should log a honeybadger error? That might get noisy though. However having all those `redirect_params` values in a HoneyBadger would probably make debugging things really quick. I wonder if we can log a honeybadger error but set them to silent unless they surpass a certain threshold.

---

There's a whole logging story that we need to follow up with in this controller (see [Jira task](https://codedotorg.atlassian.net/browse/P20-291)). Maybe we start that here, and continue with the rest of the code on a follow up PR? Or just do it all at once in a later PR?

We don't want to return a specific error message to the client here though, we've been purposeful to only return 401 Unauthorized for security reasons.

---

What determines whether or not Honeybadger logs an error? Does it generically log all errors, with the main benefit of calling the Honeybadger API that you can customize the attributes present in the error? Or does not actually not log errors unless you specifically catch and log the error?

---

Uncaught exceptions are logged to HoneyBadger. If you are catching the exception then throwing a `401` error, we would need to manually call Honeybadger.